### DISCUSSION TITLE
Partial build | Corporate project

### DISCUSSION THREAD

**Discussion by MwSpaceLLC** [vercel/next.js#59631] [Ideas]
### Goals

	1.	Reduced Compilation Time: Significantly decrease the time required for compiling large-scale Next.js projects by compiling only the modified parts.
	2.	Efficiency in Development: Enhance the development efficiency by avoiding the complete recompilation of the entire project for minor code changes.
	3.	Improved Developer Experience: Provide a smoother and more productive development experience, particularly in large projects where frequent changes are made.
	4.	Resource Optimization: Minimize the use of system resources during the development process by reducing unnecessary compilation tasks.

### Non-Goals

	1.	General Architecture Changes in Next.js: This request focuses solely on compilation; it does not propose changes to the core architecture or principal functionalities of Next.js.
	2.	Runtime Code Optimization: The objective is to improve the compilation process during development, not to optimize the performance of code execution in production.
	3.	Dependency Management or Code Versioning Solutions: This feature aims specifically at partial compilation, and does not deal with dependency management or versioning issues in source code control.

### Background

In projects with thousands of files and hundreds of pages, even minor modifications, such as changing a couple of lines of code, currently trigger a complete recompilation of the entire project. This process is time-consuming and often unnecessary, especially when the changes are localized to specific files or components.

### Proposal

Hello Next.js Team,

I’m reaching out to propose a feature that could enhance the development experience and efficiency for large-scale Next.js projects: Partial Compilation.

The Problem:
In projects with thousands of files and hundreds of pages, even minor modifications, such as changing a couple of lines of code, currently trigger a complete recompilation of the entire project. This process is time-consuming and often unnecessary, especially when the changes are localized to specific files or components.

Proposed Solution:
I would like to suggest the implementation of a partial compilation feature. The idea is to enable Next.js to compile only the parts of the project that have been modified, rather than recompiling everything. This could significantly reduce build times, making the development process faster and more efficient.

Expected Benefits:

	•	Speed: Drastically reduce compilation times for large projects.
	•	Efficiency: Save resources by avoiding unnecessary recompilation of unchanged parts.
	•	Developer Experience: Improve the overall workflow, especially in larger teams where multiple components are being worked on simultaneously.

While I understand that this might be a complex feature to implement, its potential impact on large-scale development could be substantial. It would be great to know if this is something feasible within the Next.js framework and if it aligns with your roadmap.

Thank you for considering this proposal. I look forward to any feedback or thoughts on this matter.

**Comment by umashankar-pardhi-amla** [comment]
This is must required, as I see for large projects its taking around 10-15 min to build the application and its not acceptable. For smaller applications its working fast

**Comment by dryor** [comment]
any update?
